src/scripting/scriptingService.ts

```diff
@@ -43,23 +51,62 @@ export class ScriptingService {
         7: 'SqlServerStretchEdition'
     };
 
+    /**
+     * Helper to node name from label and metadata
+     */
+    private getObjectName(nameLabel: string, objectName: string, metadata: ObjectMetadata[]): string {
+        if (nameLabel === objectName) {
+            return nameLabel;
+        }
+        // if the names aren't the same, find the closest
```

Comment:
I assume you're doing this because the labels are going to have other stuff like the `(System-Versioned)` in it?

Regardless I think there's a bug here if there's an object that has a name that matches some of the "extra" bit. For example if I have an object named `(System-Versioned)` and am searching for the metadata for a system-versioned node named `mydb` then the label will be `mydb (System-Versioned)` and it'll end up with the metadata for the `(System-Versioned)` object because the label contains that name and it's longer than the `mydb` name.

Reply:
Currently, there's no way to verify this because Object Explorer doesn't send the metadata with it. How do I go about checking this?

Reply:
By adding the metadata to the node itself as I suggested below seems like the correct way to me 😄

Reply:
This was explicitly removed from sqltoolsservice because it was making OE too slow, getting metadata with every node. That's the reason we had metadata service in the first place afaik